nixos/modules/services/networking/nar-serve.nix

```
@@ -0,0 +1,57 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nar-serve;
+in
+{
+  meta = {
+    maintainers = [ maintainers.rizary ];
+  };
+  options = {
+    services.nar-serve = {
+      enable = mkEnableOption "serve NAR file content with a webserver";
+
+      port = mkOption {
+        type = types.int;
+        default = 8383;
+        description = ''
+          Port number where nar-serve will listen on.
+        '';
+      };
+
+      nixCacheURL = mkOption {
+        type = types.str;
+        default = "https://cache.nixos.org/";
+        description = ''
+          Binary cache URL to connect to.
+
+          nar-serve also compatible with nix remote url style, such as:
+          - HttpBinaryCacheStore (http://, https://) for binary caches via HTTP or HTTPS
+          - S3BinaryCacheStore (s3://) for binary caches stored in Amazon S3
+          - LegacySSHStore (ssh://) for implement remote builds and nix-copy-closure
+          - SSHStore (ssh-ng://) for arbitrary Nix operations on a remote machine via the same protocol used by nix-daemon
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.nar-serve = {
+      description = "serve NAR file content with a webserver";
```

```suggestion
      description = "NAR Server";
```

Systemd service descriptions are used in "Starting" messages, so they they need to short noun phrases (otherwise you get messages like "Starting serve NAR file content with a webserver" which is not grammatical).